### PR TITLE
added neson design 810e keyboard

### PR DESCRIPTION
**Pull request by yulei**
add neson design 810e keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/neson_design/810e/810e.h
Comment (fauxpark) [inline keyboards/neson_design/810e/810e.h:1]: This file can be deleted.

File: keyboards/neson_design/810e/810e.c [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "810e.h"
```
Comment (fauxpark) [inline keyboards/neson_design/810e/810e.c:18]:
```suggestion
#include "quantum.h"
```

File: keyboards/neson_design/810e/info.json [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    "indicators": {
+        "caps_lock": "A10",
+        "num_lock": "A0"
+    },
+    "matrix_pins": {
+    "cols": ["B7", "B5", "B4", "B3", "A15", "A8", "B15", "B14", "B13", "B12", "B10", "B2", "B1", "B0", "A4", "A3", "A2", "A1"],
+    "rows": ["C13", "C14", "A6", "A7", "A5"]
+  },
+  "rgblight": {
+    "hue_steps": 8,
```
Comment (fauxpark) [inline keyboards/neson_design/810e/info.json:28]:
```suggestion
    "driver": "custom",
```

File: keyboards/neson_design/810e/info.json
Comment (fauxpark) [inline keyboards/neson_design/810e/info.json:1]: Please run `qmk format-json -i` on this file.

File: keyboards/neson_design/810e/readme.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Neson Design 810E keyboard
+
+Keyboard Maintainer: [astro](https://github.com/yulei)  
+Hardware Supported: Neson Design 810E keyboard
+Hardware Availability: [Public GB](https://geekhack.org/index.php?topic=120539.0)
```
Comment (fauxpark) [inline keyboards/neson_design/810e/readme.md:5]:
```suggestion
* Keyboard Maintainer: [astro](https://github.com/yulei)
* Hardware Supported: Neson Design 810E keyboard
* Hardware Availability: [Public GB](https://geekhack.org/index.php?topic=120539.0)
```

File: keyboards/neson_design/810e/readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make neson_design/810e:default
```
Comment (fauxpark) [inline keyboards/neson_design/810e/readme.md:9]:
```suggestion
    make neson_design/810e:default

Flashing example for this board:

    make neson_design/810e:default:flash
```

File: keyboards/neson_design/810e/rules.mk [hunk 5]
```diff
@@ -0,0 +1,5 @@
+
+NO_USB_STARTUP_CHECK = yes
+
+SRC += is31fl3236.c is31fl3729.c
+QUANTUM_LIB_SRC += i2c_master.c
```
Comment (fauxpark) [inline keyboards/neson_design/810e/rules.mk:5]:
```suggestion
I2C_DRIVER_REQUIRED = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/neson_design/810e/mcuconf.h [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+#pragma once
+
+#include_next <mcuconf.h>
+
+#undef STM32_I2C_USE_I2C1
+#define STM32_I2C_USE_I2C1  TRUE
+#define I2C1_SCL_PIN B8
+#define I2C1_SDA_PIN B9
```
Comment (drashna) [inline keyboards/neson_design/810e/mcuconf.h:24]: These would belong in the config.h, not here. 
```suggestion
```

File: keyboards/neson_design/810e/config.h [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define BOARD_OTG_NOVBUSSENS 1
+#define STM32_HSECLK 16000000U
+
+#define IS31FL3729_LED_BEGIN    0
```
Comment (drashna) [inline keyboards/neson_design/810e/config.h:23]:
```suggestion
#define I2C1_SCL_PIN B8
#define I2C1_SDA_PIN B9

#define IS31FL3729_LED_BEGIN    0
```

File: keyboards/neson_design/810e/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/neson_design/810e/keymaps/default/keymap.c:1]: Please replace tabs with spaces

File: keyboards/neson_design/810e/keymaps/via/keymap.c
Comment (drashna) [inline keyboards/neson_design/810e/keymaps/via/keymap.c:1]: Please replace the tabs with spaces

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by yulei** [comment]
@drashna anything blocking the review process?

File: keyboards/neson_design/810e/info.json [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    },
+    "processor": "STM32F411",
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x810E",
+        "vid": "0x4E65"
+    },
+    "layouts": {
+        "LAYOUT_default": {
```
Comment (waffle87) [inline keyboards/neson_design/810e/info.json:30]:
```suggestion
        "LAYOUT": {
```

File: keyboards/neson_design/810e/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0]=LAYOUT_default(
+        QK_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL,          KC_BSPC,   KC_INS, KC_NUM, KC_PSLS, KC_PAST, KC_PMNS,
+        KC_TAB,     KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC,      KC_BSLS,   KC_DEL,  KC_P7,   KC_P8,   KC_P9,
+        KC_CAPS,   KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,              KC_ENT,            KC_P4,   KC_P5,   KC_P6, KC_PPLS,
+        KC_LSFT,      KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH,   KC_RSFT,               KC_UP,   KC_P1,   KC_P2,   KC_P3,
+        KC_LCTL, KC_LGUI, KC_LALT,                 KC_SPC,                    KC_RALT,MO(1),KC_RCTL,  KC_LEFT,   KC_DOWN, KC_RGHT,   KC_P0, KC_PDOT, KC_PENT),
+
+[1]=LAYOUT_default(
```
Comment (waffle87) [inline keyboards/neson_design/810e/keymaps/default/keymap.c:26]:
```suggestion
    [0]=LAYOUT(
        QK_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL,          KC_BSPC,   KC_INS, KC_NUM, KC_PSLS, KC_PAST, KC_PMNS,
        KC_TAB,     KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC,      KC_BSLS,   KC_DEL,  KC_P7,   KC_P8,   KC_P9,
        KC_CAPS,   KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,              KC_ENT,            KC_P4,   KC_P5,   KC_P6, KC_PPLS,
        KC_LSFT,      KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH,   KC_RSFT,               KC_UP,   KC_P1,   KC_P2,   KC_P3,
        KC_LCTL, KC_LGUI, KC_LALT,                 KC_SPC,                    KC_RALT,MO(1),KC_RCTL,  KC_LEFT,   KC_DOWN, KC_RGHT,   KC_P0, KC_PDOT, KC_PENT),

[1]=LAYOUT(
```

File: keyboards/neson_design/810e/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______,
+        _______, RGB_TOG, RGB_MOD,  KC_F13, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,
+        RESET, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                   _______, _______, _______, _______,
```
Comment (waffle87) [inline keyboards/neson_design/810e/keymaps/default/keymap.c:29]: `RESET` has been deprecated for a very long time now. Please be testing your changes.
```suggestion
        QK_BOOT, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                   _______, _______, _______, _______, 
```

File: keyboards/neson_design/810e/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0]=LAYOUT_default(
+		QK_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL,          KC_BSPC,   KC_INS, KC_NUM, KC_PSLS, KC_PAST, KC_PMNS,
+		KC_TAB,     KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC,      KC_BSLS,   KC_DEL,  KC_P7,   KC_P8,   KC_P9,
+		KC_CAPS,   KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,              KC_ENT,            KC_P4,   KC_P5,   KC_P6, KC_PPLS,
+		KC_LSFT,      KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH,   KC_RSFT,               KC_UP,   KC_P1,   KC_P2,   KC_P3,
+		KC_LCTL, KC_LGUI, KC_LALT,                 KC_SPC,                    KC_RALT,MO(1),KC_RCTL,  KC_LEFT,   KC_DOWN, KC_RGHT,   KC_P0, KC_PDOT, KC_PENT),
+
+[1]=LAYOUT_default(
```
Comment (waffle87) [inline keyboards/neson_design/810e/keymaps/via/keymap.c:27]:
```suggestion
[0]=LAYOUT(
		QK_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL,          KC_BSPC,   KC_INS, KC_NUM, KC_PSLS, KC_PAST, KC_PMNS,
		KC_TAB,     KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC,      KC_BSLS,   KC_DEL,  KC_P7,   KC_P8,   KC_P9,
		KC_CAPS,   KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,              KC_ENT,            KC_P4,   KC_P5,   KC_P6, KC_PPLS,
		KC_LSFT,      KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH,   KC_RSFT,               KC_UP,   KC_P1,   KC_P2,   KC_P3,
		KC_LCTL, KC_LGUI, KC_LALT,                 KC_SPC,                    KC_RALT,MO(1),KC_RCTL,  KC_LEFT,   KC_DOWN, KC_RGHT,   KC_P0, KC_PDOT, KC_PENT),

[1]=LAYOUT(
```

File: keyboards/neson_design/810e/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______,
+		_______, RGB_TOG, RGB_MOD,  KC_F13, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,
+		RESET, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                   _______, _______, _______, _______,
```
Comment (waffle87) [inline keyboards/neson_design/810e/keymaps/via/keymap.c:30]:
```suggestion
		QK_BOOT, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                   _______, _______, _______, _______, 
```

File: keyboards/neson_design/810e/mcuconf.h [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include_next <mcuconf.h>
+
+#undef STM32_I2C_USE_I2C1
+#define STM32_I2C_USE_I2C1  TRUE
```
Comment (waffle87) [inline keyboards/neson_design/810e/mcuconf.h:22]: What exactly is I2C used for on this board?

File: keyboards/neson_design/810e/info.json [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+    "matrix_pins": {
+        "cols": ["B7", "B5", "B4", "B3", "A15", "A8", "B15", "B14", "B13", "B12", "B10", "B2", "B1", "B0", "A4", "A3", "A2", "A1"],
+        "rows": ["C13", "C14", "A6", "A7", "A5"]
+    },
+    "processor": "STM32F411",
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x810E",
+        "vid": "0x4E65"
```
Comment (waffle87) [inline keyboards/neson_design/810e/info.json:27]:
```suggestion
        "vid": "0x4E65",
        "no_startup_check": true
```

File: keyboards/neson_design/810e/rules.mk [hunk 15]
```diff
@@ -0,0 +1,2 @@
+
+NO_USB_STARTUP_CHECK = yes
```
Comment (waffle87) [inline keyboards/neson_design/810e/rules.mk:2]:
```suggestion
# This file intentionally left blank
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/neson_design/810e/mcuconf.h (hunk 13 above)
Comment (yulei) [reply keyboards/neson_design/810e/mcuconf.h]: i2c related codes were removed

**Review by fauxpark — APPROVED** [review]

File: keyboards/neson_design/810e/keymaps/via/keymap.c [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+		KC_LCTL, KC_LGUI, KC_LALT,                 KC_SPC,                    KC_RALT,MO(1),KC_RCTL,  KC_LEFT,   KC_DOWN, KC_RGHT,   KC_P0, KC_PDOT, KC_PENT),
+
+[1]=LAYOUT(
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______,
+		_______, RGB_TOG, RGB_MOD,  KC_F13, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,
+		QK_BOOT, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                   _______, _______, _______, _______, 
+		_______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,           KC_VOLU,_______, _______, _______,
+		_______, _______, _______,                            KC_MPLY,                                     _______, _______, _______, KC_MPRV, KC_VOLD, KC_MNXT, _______, _______, _______)
```
Comment (drashna) [inline keyboards/neson_design/810e/keymaps/via/keymap.c:32]:
```suggestion
        QK_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL,          KC_BSPC,   KC_INS, KC_NUM, KC_PSLS, KC_PAST, KC_PMNS,
        KC_TAB,     KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC,      KC_BSLS,   KC_DEL,  KC_P7,   KC_P8,   KC_P9,
        KC_CAPS,   KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,              KC_ENT,            KC_P4,   KC_P5,   KC_P6, KC_PPLS,
        KC_LSFT,      KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH,   KC_RSFT,               KC_UP,   KC_P1,   KC_P2,   KC_P3,
        KC_LCTL, KC_LGUI, KC_LALT,                 KC_SPC,                    KC_RALT,MO(1),KC_RCTL,  KC_LEFT,   KC_DOWN, KC_RGHT,   KC_P0, KC_PDOT, KC_PENT),

[1]=LAYOUT(
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______, _______,
        _______, RGB_TOG, RGB_MOD,  KC_F13, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,
        QK_BOOT, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                   _______, _______, _______, _______, 
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,           KC_VOLU,_______, _______, _______,
        _______, _______, _______,                            KC_MPLY,                                     _______, _______, _______, KC_MPRV, KC_VOLD, KC_MNXT, _______, _______, _______)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**